runtime/compiler/env/J9SharedCache.cpp

```diff
@@ -96,6 +100,45 @@ TR_YesNoMaybe TR_J9SharedCache::isSharedCacheDisabledBecauseFull(TR::Compilation
    return _sharedCacheDisabledBecauseFull;
    }
 
+bool
+TR_J9SharedCache::initCCVCaching()
+   {
+   if (!_classChainValidationMutex)
+      {
+      if (!(_classChainValidationMutex = TR::Monitor::create("JIT-ClassChainValidationMutex")))
+         return false;
+      }
+
+   void *storage = jitPersistentAlloc(sizeof(CCVMap));
+   if (!storage)
+      return false;
+
+   _ccvMap = new (storage) CCVMap(CCVComparator(), TR::Compiler->persistentAllocator());
```

Comment:
Probably should guard this with `if (!_ccvMap`. Or, I should just remove the `if (!_classChainValidationMutex)` check above.

Reply:
Would something from `PersistentCollections.hpp` work for you here? (maybe extend it as I see that we only have  unordered_map)

Reply:
I _think_ that the syntax in that file won't compile on all platforms; it needs full C++11 support.